src/prefect/engine.py

```diff
@@ -1256,16 +1256,29 @@ def get_state_for_result(obj: Any) -> Optional[State]:
 
 def link_state_to_result(state: State, result: Any) -> None:
     """
-    Stores information about the state on the result or in the global context for
-    relationship tracking.
+    Caches a link between a state and a result using the `id` of the result to map to
+    the state. The cache is persisted to the current flow run context since task
+    relationships are limited to within a flow run.
+
+    This allows dependency tracking to occur when results are passed around.
+
+    We do not hash the result because:
```

Comment:
```suggestion
    We do not hash the result because:
    
```

Reply:
Need a blank line before a bullet list, otherwise it gets pulled up into the preceding paragraph.